lib/assets/javascripts/twine.coffee

```diff
@@ -46,33 +46,47 @@
     this
 
   Twine.bind = (node = rootNode, context = Twine.context(node)) ->
-    bind(context, node, true)
+    bind(context, node, getIndexesForElement(node), true)
 
   Twine.afterBound = (callback) ->
     if currentBindingCallbacks
       currentBindingCallbacks.push(callback)
     else
       callback()
 
-  bind = (context, node, forceSaveContext) ->
+  bind = (context, node, indexes, forceSaveContext) ->
     currentBindingCallbacks = []
     if node.bindingId
       Twine.unbind(node)
 
+    if defineArrayAttr = Twine.getAttribute(node, 'define-array')
+      newIndexes = defineArray(node, context, defineArrayAttr)
+      for key, value of indexes || {} when !newIndexes[key]?
```

Comment:
I don't want to do indexes as a default arg because that would mean a lot of elements storing an empty object as the indexes, which means we have to check the number of keys in an object to tell if an element has indexes rather than just a null check.

Better to keep it as null/undefined when there are no indexes to store.